drivers/hmis/app/graphql/types/base_audit_event.rb

```diff
@@ -110,6 +111,11 @@ def graphql_type
       end
     end
 
+    private def custom_assessment_title
+      ca = load_ar_scope(scope: Hmis::Hud::CustomAssessment.with_deleted, id: object.item_id)
+      ca&.form_processor&.definition&.title
```

Comment:
This will create an N+1 I believe, because each audit even in the table will be joining to form processor and definition separately. I'm not sure if there is a way to avoid it in this case though. It sounded like @ttoomey  had some ideas there?

Reply:
To avoid n+1 issues in graphql, you should be able to use the data loader pattern rather than the normal active record associations. Looks like CustomAssessment has a relationship to definition so we can do something like this:

```suggestion
      ca ? load_ar_association(object, :definition)&.title : nil
```

Reply:
Something is funny with this -- I think because `object` is a `GrdaWarehouse::Version`, and not a `CustomAssessment` ? 
```
Association named 'definition' was not found on GrdaWarehouse::Version; perhaps you misspelled it?
```

Reply:
oh shoot sorry! I think that should `ca` instead of `object`- something like:
```ruby
ca ? load_ar_association(ca, :definition)&.title : nil
```

Reply:
I've updated it to use the `changed_record` from line 120 (just below), let me know if you think that is correct

Reply:
Ah no you're totally right, it should be `ca` and not `changed_record`, because `changed_record` will return nil if the item has been soft deleted.

I was thinking since these are audit events it might be nice to change that, and end up with something like this:
```
private def custom_assessment_title
  return unless changed_record.is_a?(Hmis::Hud::CustomAssessment)
  
  load_ar_association(changed_record, :definition)&.title
end

private def changed_record
  load_ar_association(object, :item, scope: with_deleted)
end
```
but the with_deleted scope does not work here because not all of the objects we might be auditing act as paranoid.

(Anyway, this seems like it may be scope creep 🦂)

Reply:
Hmm, you  might be able to achieve something like that by introspecting on the item.owner_type to see if the scope should use `with_deleted`. Something like this: 
`object.item_type.constantize.paranoid?`

I'm not sure it's really getting us much here though; it seems okay to leave as it is

Reply:
Yes, right now it wouldn't be getting us anything since the FormProcessor models don't act as paranoid anyway! But I added a comment on this [PT issue](https://www.pivotaltracker.com/n/projects/2591838/stories/187150766) so we remember to come back to it when we make that change